torchtitan/checkpoint.py

```diff
@@ -106,6 +107,12 @@ class SaveDone:
     pass
 
 
+@torch.no_grad()
+def save_with_gc(state, checkpoint_id):
+    dcp.save(state, checkpoint_id=checkpoint_id)
+    gc.collect(1)
```

Comment:
remind me, why (1)? first generation? means only collect easy to collect objects, or something?

Reply:
Yes only first generation. It has been proven to be enough for avoiding OOM for training. It is also faster than the default one.